packages/tempus/src/Tempus_StepperHHTAlpha_impl.hpp

```diff
@@ -233,8 +233,19 @@ void StepperHHTAlpha<Scalar>::takeStep(
       RCP<Thyra::VectorBase<Scalar> > a_init = Thyra::createMember(a_old->space());
       Thyra::copy(*d_old, d_init.ptr());
       Thyra::copy(*v_old, v_init.ptr());
-      Thyra::put_scalar(0.0, a_init.ptr());
-      wrapperModel->initializeNewmark(a_init,v_init,d_init,0.0,time,beta_,gamma_);
+      if (initial_guess_ != Teuchos::null) { //set initial guess for Newton, if provided
+        //Throw an exception if initial_guess is not compatible with solution 
+        bool is_compatible = (a_init->space())->isCompatible(*initial_guess_->space()); 
```

Comment:
The test really should be "isSameAs" in Tpetra terms, since two Maps can be compatible but locally a permutation of each other.  Not sure how Thyra expresses that idea.  Also, note that compatibility and sameness tests for Maps may cost an all-reduce in the worst case.